src/main/java/com/greghaskins/spectrum/internal/LeafChild.java

```diff
@@ -0,0 +1,14 @@
+package com.greghaskins.spectrum.internal;
+
+import com.greghaskins.spectrum.internal.hooks.Hook;
+
+/**
+ * Interface for a child that is also a leaf node in the test hierarchy.
+ */
+public interface LeafChild extends Child {
```

Comment:
This _leaf_ interface allows us to use `instanceof` and a cast to specific type in order to work with a leaf node to apply leaf-level hooks. If we didn't do that we'd end up adding an unreachable method to `Child` for some of its descendants - the code coverage led me to this approach.